Islandora/islandora_defaults
# Fixing facets that are breaking search

**GitHub Issue**: Follow up to https://github.com/Islandora/islandora_defaults/pull/52

Blocking https://github.com/Islandora/islandora/pull/843

# What does this Pull Request do?

Points facets towards the correct fields on the solr index after the MIG sandbox changes. The facets are currently pointing at fields that are no longer in the index, and it's making search fail silently.

# What's new?

* Touching up the solr index fields
* Pointing facets towards those fields instead of nonexistant ones
* Deleted vocabs here that already are / should be in controlled_access_terms_defaults.

# How should this be tested?

Original Behaviour:

Try to search, you'll get no results and no real indication that anything went wrong.

## How to get an environment to test this

I used ISLE. Make sure you have the latest `isle-dc`, and from within its directory

1. `sudo rm -rf codebase; git clone https://github.com/islandora-devops/islandora-sandbox codebase`
2. Set `ENVIRONMENT=local` in your .env (rename `sample.env` to `.env` if you don't have it)
3. Set `TAG=1.0.0-alpha-9`
3. `make local`
4. Pull ths PR into `codebase/web/modules/contrib/islandora_defaults`
5. Pull https://github.com/Islandora/controlled_access_terms/pull/71 into `codebase/web/modules/contrib/controlled_access_terms`
6. `docker-compose exec drupal with-contenv bash -lc 'drush -y en facets name islandora_search'`
9. `docker-compose exec drupal with-contenv bash -lc 'drush -y fim controlled_access_terms_defaults'`
10. `docker-compose exec drupal with-contenv bash -lc 'drush -y fim islandora_defaults'`
11. `docker-compose exec drupal with-contenv bash -lc 'drush -y fim islandora_search'` (it will complain about not being able to reach the solr server.  that's fine.  the next step fixes it)
12. `make hydrate`

Then you can go add some terms, make some content, and try to search it.  Search should work and you should see facets.

![image](https://user-images.githubusercontent.com/20773151/126814808-be8db362-c7d8-4f29-941d-a33fe3cd7051.png)

# Interested parties

@Islandora/8-x-committers
